uefi-mkconfig

```diff
@@ -198,58 +157,109 @@ main () {
 
 	# Verify kernel commands
         for check_config in $kernel_commands; do
-
                 # Strip initrd=* from config file
                 [[ "$check_config" != *"initrd="* ]] && strip_kernel_commands="$strip_kernel_commands $check_config"
                 # Verify if "root=" is present
                 [[ "$check_config" == "root="* ]] && valid_kernel_commands=1
-
         done
 	kernel_commands="$strip_kernel_commands"
 
         [[ -z $valid_kernel_commands ]] && ewarn "Warning, kernel command \"root=\" is missing from loaded configuration!"
+	
+	# Get initial state of UEFI entries
+	initial_uefi_state="$(efibootmgr -u)"
+	
+	# Wipe all managed entries for regeneration
+	wipe_entries
+	
+	partition_mounts="${mounted_efi_partitions// "$efi_parttype"/}"
+	
+	# Find all .efi files on this partition and split them to old and new
+	for prefix in $kernel_prefixes; do
+		find_results_new="$(find $partition_mounts -name "$prefix*.efi" ! -name "$prefix*-old.efi" -printf "%f\n")"
+		find_results_old="$(find $partition_mounts -name "$prefix*-old.efi" -printf "%f\n")"
+		[[ -n "$find_results_new" ]] && kernel_images_new+="$find_results_new "
+		[[ -n "$find_results_old" ]] && kernel_images_old+="$find_results_old "
```

Comment:
Here we append to these variables, so we should ensure they are empty when we start `main`